webapp/src/ts/modals/guided-setup/guided-setup.component.ts

```diff
@@ -76,21 +78,28 @@ export class GuidedSetupComponent extends MmModalAbstract implements AfterViewIn
       (<any>$('#guided-setup [name=default-country-code]')).select2({ width: '20em', data: this.countryList });
       $('#guided-setup [name=default-country-code]').on('change', this.updateNumbers);
       this.settingsService.get().then((res: any) => {
+        this.settingsLoaded = true;
+        this.settings = res;
         if (res.setup_complete) {
-          $('#guided-setup [name=gateway-number]').val(res.gateway_number).trigger('input');
-          $('#primary-contact-content a[data-value=' + res.care_coordinator + ']').trigger('click');
-          $('#registration-form-content a[data-value=' + res.anc_registration_lmp + ']').trigger('click');
-          setTimeout(() => { // setTimeout used to make sure ngFor generated list is ready
-            $('#guided-setup [name=default-country-code]').val(res.default_country_code).change();
-            $('#language-preference-content .locale a[data-value=' + res.locale + ']').trigger('click');
-            $('#language-preference-content .locale-outgoing a[data-value=' + res.locale_outgoing + ']')
-              .trigger('click');
-          }, 10);
+          $('#guided-setup [name=gateway-number]').val(this.settings.gateway_number).trigger('input');
+          $('#primary-contact-content a[data-value=' + this.settings.care_coordinator + ']').trigger('click');
+          $('#registration-form-content a[data-value=' + this.settings.anc_registration_lmp + ']').trigger('click');
         }
       })
     });
   }
 
+  ngAfterContentChecked() {
+    if (this.settingsLoaded) {
```

Comment:
Also, because you're not setting the flag to false, this runs every time when something changes and the result is that you can't select any other language.